src/devices_models/devices/common/activepower_constraints.jl

```diff
@@ -0,0 +1,41 @@
+"""
+Construct active power DeviceRangeConstraintInputs for specific types.
+"""
+function make_active_power_constraints_inputs(
+    ::Type{T},
+    ::Type{U},
+    ::Type{V},
+    feedforward::Union{Nothing, AbstractAffectFeedForward},
+    use_parameters::Bool,
+    use_forecasts::Bool,
+) where {T <: PSY.Device, U <: AbstractDeviceFormulation, V <: PM.AbstractPowerModel}
+    error("make_active_power_constraints_inputs is not implemented for types $T / $U / $V")
```

Comment:
`ArgumentError`?

Reply:
I like a hard error with this simple explanation.  `ArgumentError` from Julia would make this look like a programming mistake.  Maybe it is?  I'd vote for the simple `error()`.

Reply:
ok that's fine